drivers: memc: {stm32_xspi_psram,stm32_ospi_psram}: make NCS boundary configurable in dt

arthur-mmlw opened the pull request:
Limit a transaction to a boundary of aligned addresses. Each PSRAM may specify different configuration.

Reference: https://www.st.com/resource/en/reference_manual/rm0477-stm32h7rx7sx-armbased-32bit-mcus-stmicroelectronics.pdf#page=1016

arthur-mmlw committed:
drivers: memc: stm32_ospi_psram: make NCS boundary configurable in dt

Limit a transaction to a boundary of aligned addresses. Each PSRAM may
specify different configuration.

Signed-off-by: Arthur Gay <arthur.gay@marshmallow.kids>

arthur-mmlw committed:
drivers: memc: stm32_xspi_psram: make NCS boundary configurable in dt

Limit a transaction to a boundary of aligned addresses. Each PSRAM may
specify different configuration.

Signed-off-by: Arthur Gay <arthur.gay@marshmallow.kids>

erwango reviewed (commented):
Thanks!

For consistency, this should probably be also added to recently merged `st,stm32-ospi-psram` as well which was merged earlier today.
This also raises the question of factorizing part of the binding, but of the scope of this PR.

arthur-mmlw commented on dts/bindings/memory-controllers/st,stm32-xspi-psram.yaml:
```diff
@@ -146,3 +146,13 @@ properties:
+
+  st,csbound:
+    type: int
+    default: 11
```
The optimal value is specific to each board, as it depends on the external PSRAM mounted on the board.

It is not app-specific. Once the PSRAM is mapped to the internal address space, the app uses the PSRAM transparently, regardless of the page size.

arthur-mmlw replied:
Should I keep the default value in the bindings or mark the property as required then?

arthur-mmlw committed:
boards: st: Set csbounds explicitly instead of relying on default value

The default value has changed in previous commits, so set it explicitly
in the devicetree with the new st,csbound property.

Signed-off-by: Arthur Gay <arthur.gay@marshmallow.kids>

etienne-lms reviewed (approved):
CI build test failure is unrelated, see https://github.com/zephyrproject-rtos/zephyr/pull/96635.
